src/Listener/WebhookListener.php

```diff
@@ -0,0 +1,166 @@
+use ZfrCash\Service\PlanService;
+use ZfrCash\Service\SubscriptionService;
+use ZfrStripe\Client\StripeClient;
+
+/**
+ * Listener that synchronizes Stripe data with our own database data
+ *
+ * @author Michaël Gallego <mic.gallego@gmail.com>
+ */
+final class WebhookListener extends AbstractListenerAggregate
+{
+    /**
+     * @var ServiceLocatorInterface
+     */
+    private $serviceLocator;
+
+    /**
+     * @var ModuleOptions
+     */
+    private $moduleOptions;
+
+    /**
+     * Constructor
+     *
+     * NOTE: while this is not the best practice to inject the whole service locator, we mainly do that for
+     * performance reasons, because this listener may use a lot of different services depending on the event type
+     * received, and we do not want to create tons of services at each requests for nothing
+     *
+     * @param ServiceLocatorInterface $serviceLocator
+     */
+    public function __construct(ServiceLocatorInterface $serviceLocator)
+    {
+        $this->serviceLocator = $serviceLocator;
+        $this->moduleOptions  = $serviceLocator->get(ModuleOptions::class);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function attach(EventManagerInterface $eventManager)
+    {
+        $sharedManager = $eventManager->getSharedManager();
+        $sharedManager->attach(WebhookListenerController::class, WebhookEvent::WEBHOOK_RECEIVED, [$this, 'dispatchWebhook']);
+    }
+
+    /**
+     * @internal
+     * @param  WebhookEvent $event
+     * @return string
+     */
+    public function dispatchWebhook(WebhookEvent $event)
+    {
+        switch ($event->getEventName()) {
+            case 'customer.discount.created':
+            case 'customer.discount.updated':
+            case 'customer.discount.deleted':
+                return $this->handleDiscountEvent($event->getEvent());
+
+            case 'customer.subscription.updated':
+            case 'customer.subscription.deleted':
+                return $this->handleSubscriptionEvent($event->getEvent());
+
+            case 'plan.created':
+            case 'plan.updated':
+            case 'plan.deleted':
+                return $this->handlePlanEvent($event->getEvent());
+
+            case 'invoice.created':
+            case 'invoice.updated':
+            case 'invoice.payment_succeeded':
+            case 'invoice.payment_failed':
+                return $this->handleInvoiceEvent($event->getEvent());
+
+            default:
+                return ''; // Any other event is not handled by default
+        }
+    }
+
+    /**
+     * Handle a discount Stripe event
+     *
+     * @param  array $stripeEvent
+     * @return string
+     */
+    public function handleDiscountEvent(array $stripeEvent)
+    {
+        /** @var DiscountService $discountService */
+        $discountService = $this->serviceLocator->get(DiscountService::class);
+        $discountService->syncFromStripeEvent($stripeEvent);
+
+        return 'Event has been properly processed';
```

Comment:
Yeah, sorry for this not being clear for you. It's actually processed elsewhere, and all the string answers are concatenated and sent to the response. Basically having human readable string allows to debug events in Stripe interface, to check if they have been correctly received and processed.